search/api.py

```diff
@@ -0,0 +1,261 @@
+"""
+Functions for search
+"""
+from itertools import islice
+import logging
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from elasticsearch.helpers import bulk
+from elasticsearch.exceptions import NotFoundError
+from elasticsearch_dsl import Mapping
+from elasticsearch_dsl.connections import connections
+
+from profiles.models import Profile
+from profiles.serializers import ProfileSerializer
+from search.exceptions import ReindexException
+
+log = logging.getLogger(__name__)
+
+
+USER_DOC_TYPE = 'user'
+DOC_TYPES = (USER_DOC_TYPE, )
+_CONN = None
+# When we create the connection, check to make sure all appropriate mappings exist
+_CONN_VERIFIED = False
```

Comment:
I don't think it's worth encapsulating into `ESConnection` since we still need one global variable to store the `ESConnection`. Both variables are only used by `get_conn`